synthtool/languages/python.py

```diff
@@ -148,6 +148,58 @@ def py_samples(*, root: PathOrStr = None, skip_readmes: bool = False) -> None:
         s.copy([result], excludes=excludes)
 
 
+def configure_previous_major_version_branches() -> None:
+    """Configure releases from previous major version branches by editing
+     `.github/release-please.yml`.
+
+     The current library version is obtained from `version.py` in `google/**/version.py`,
+     or the `setup.py`.
+
+     Releases are configured for all previous major versions. For example,
+     if the library version is currently 3.5.1, the release-please config
+     will include v2, v1, and v0.
+     """
+    if list(Path(".").glob("google/**/version.py")):
+        version_file = list(Path(".").glob("google/**/version.py"))[0]
+    else:
+        version_file = Path("setup.py")
+
+    # In version.py:    __version__ = "1.5.2"
+    # In setup.py:      version = "1.5.2"
+    VERSION_REGEX = (
+        r"(?:__)?version(?:__)?\s*=\s*[\"'](?P<major_version>\d)\.[\d\.]+[\"']"
+    )
+
+    match = re.search(VERSION_REGEX, Path(version_file).read_text())
+
+    if match is not None:
+        major_version = int(match.group("major_version"))
+    else:
+        raise RuntimeError(
+            "Unable to find library version in {} with regex {}".format(
+                version_file, VERSION_REGEX
+            )
+        )
+
+    if major_version > 0:
+        branch_release_config = """# NOTE: this section is generated by synthtool.languages.python
+# See https://github.com/googleapis/synthtool/blob/master/synthtool/languages/python.py
+branches:"""
+
+        for v in range(major_version - 1, -1, -1):
+            branch_release_config += """
+  - branch: v{}
+    handleGHRelease: true
+    releaseType: python""".format(
+                v
+            )
+
+        release_please_yml = Path(".github/release-please.yml").read_text()
+        Path(".github/release-please.yml").write_text(
+            release_please_yml + branch_release_config
+        )
```

Comment:
Is this idempotent? It looks like it will just keep appending to the file.

Also, it might be cleaner to build it as a dict and export to yaml.